packages/connectkit/src/utils/wallets.ts

```diff
@@ -0,0 +1,121 @@
+export const isMetaMask = () => {
+  if (typeof window === 'undefined') return false;
+  if (!window?.ethereum?.isMetaMask) return false;
+
+  if (isBrave()) return false;
+  if (isDawn()) return false;
+  if (isTokenary()) return false;
+  if (isFrame()) return false;
+  if (isInfinityWallet()) return false;
+  if (isRabby()) return false;
+  if (isTokenPocket()) return false;
+  if (isTalisman()) return false;
+  if (isPhantom()) return false;
```

Comment:
This line is the cause for several open issues in this repo from the current `isMetamask` implementation and the same issue is seen on the preview branches here since the offending line of code remains unchanged.

The isPhantom check ensures that as long as Phantom is installed, ConnectKit will never register metamask as being installed.

If you go into Phantom's settings while on the Preview deployment, selecting `Always use Phantom` will show the Phantom Button on the Connector List. Regardless of which option you select, the ConnectKit modal will never register Metamask as being connected.

This is also the case with all of the the similar issues w/ Brave that are seen elsewhere.

I'd advocate for removing these checks from the `isMetamask` function or refactor the logic on showing the Metamask Not Installed UI.

This is a bad user experience and although I love ConnectKit, I unfortunately can't use this in prod until it's resolved.

Reply:
I'm attaching a video showing what I'm talking about.


https://github.com/family/connectkit/assets/5373129/1b4f4f89-2107-42c3-b6f4-d110fcc9b43d

Before this PR, if you had Phantom Default App Wallet Setting as Always ask, at least you'd get a browser wallet button to make metamask work, however this new branch does away with that without giving the metamask button back its functionality

![Screenshot 2023-07-28 at 1 33 52 PM](https://github.com/family/connectkit/assets/5373129/3a6fa149-a840-4c74-a880-5a2e14c7f14b)

LMK if you'd like me to open up a branch PR or anything!

Reply:
@wcjohnson11 thank you for investigating further here, I'd really appreciate you creating a PR if possible.

Reply:
I've updated this branch to remove the override checks. The reason that was in there was because some injected wallets would override MetaMask functionality and make that button resolve to their wallet. The idea was to block that functionality and rely on wagmi's `InjectedConnector` to fall back to, but it seems like some wallets have done some more workarounds and different UX that we originally had catered for (i.e Phantom's default wallet switcher).

Have a [test of of the latest version of this branch](https://connectkit-testbench-git-injected-refactor-lfe.vercel.app/), and you'll see what I mean with how some injected wallets will override MetaMask, but let me know if there is any further problems with this new setup.

Reply:
awesome, thanks @lochie !

Reply:
Any feedback on this? If we're happy I'll get this moving ahead.

Reply:
So my personal testing has been great as I've always been able to connect using the MetaMask button, but it doesn't seem to detect Phantom wallet now if I am correct, maybe it was just something on my end. I am focusing on MetaMask users anyway. 😆 


I have forwarded the test deployment link to a customer to see if they are now happy with how the library handles the connecting but I'm waiting for a solid reply, I will see if I can reach them today.

Reply:
I'm able to get Phantom to display with the below setting, but the "Always Ask" setting does not display Phantom (expected) which I believe is because Phantom doesn't inject the same provider information into the browser window until confirmation, I would assume that Phantom would override MM in this case and ask a prompt like CBW does?

![CleanShot —08-07 at 23 38 19](https://github.com/family/connectkit/assets/1930210/5fd902eb-b93a-4ffd-9f38-59cf0e1d2c4b)

Reply:
ahh yeah you have to set the Default App Wallet to "Phantom" for it to show, but if you do, both the Phantom and MetaMask button open Phantom - which you can't do nothing about I suppose as Phantom then overrides the provider. All in all, I'm fine with how it's handled.

Reply:
We're ready to move forward with this PR unless anyone has any further feedback 🫡